curves/src/bls12_377/fr.rs

```diff
@@ -55,21 +55,12 @@ impl FftParameters for FrParameters {
 
     #[rustfmt::skip]
     const TWO_ADICITY: u32 = 47;
-    /// TODO (howardwu): CRITICAL - Fix this after a migration plan has been determined.
-    ///  - 0x3c3d3ca739381fb2,
-    ///  - 0x9a14cda3ec99772b,
-    ///  - 0xd7aacc7c59724826,
-    ///  - 0xd1ba211c5cc349c,
-    ///  + 12646347781564978760u64,
-    ///  + 6783048705277173164u64,
-    ///  + 268534165941069093u64,
-    ///  + 1121515446318641358u64,
     #[rustfmt::skip]
     const TWO_ADIC_ROOT_OF_UNITY: BigInteger = BigInteger([
-        0x3c3d3ca739381fb2,
-        0x9a14cda3ec99772b,
-        0xd7aacc7c59724826,
-        0xd1ba211c5cc349c,
+        12646347781564978760u64,
+        6783048705277173164u64,
+        268534165941069093u64,
+        1121515446318641358u64,
```

Comment:
This is 22?

Reply:
The `GENERATOR` is 22, however the `TWO_ADIC_ROOT_OF_UNITY` is 8065159656716812877374967518403273466521432693661810619979959746626482506078.

Here is the Sage output:
```
2-adic gen (g2 % q):  8065159656716812877374967518403273466521432693661810619979959746626482506078
2-adic gen (g2 * R % q):  7039866554349711480672062101017509031917008525101396696252683426045173093960
2-adic gen into_chunks(g2 * R % q):  [12646347781564978760, 6783048705277173164, 268534165941069093, 1121515446318641358]
```

Reply:
I've added comments to the const declaration to clarify this